File: Sort characters by frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<char,int> mp;
        for(auto c:s)
        {
            mp[c]++;
        }
        vector<pair<int,int>> v(mp.begin(),mp.end());
        string ans="";
        sort(v.begin(),v.end(),[](pair<int,int> a ,pair<int,int> b){
            return a.second > b.second;
        });
        for(auto k : v){
            while(k.second--){
                ans+=k.first;
            }
        }
        return ans;
    }
};

File: Sort characters by frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        d={}
        for c in s:
            if c not in d:
                d[c]=1
            else:
                d[c]+=1
        sort_d=dict(sorted(d.items(),key=operator.itemgetter(1),reverse=True))
        ans=""
        for k,v in sort_d.items():
            ans+=k*v
        return ans
        
